pkg/vulnsrc/vulnerability/vulnerability.go

```diff
@@ -211,12 +214,15 @@ func getLastModifiedDate(vulnID string, details map[types.SourceID]types.Vulnera
 }
 
 func getRejectedStatus(details map[types.SourceID]types.VulnerabilityDetail) bool {
+	if nvd, ok := details[NVD]; ok && nvd.VulnStatus == rejectedStatus {
+		return true
+	}
 	for _, source := range AllSourceIDs {
 		d, ok := details[source]
 		if !ok {
 			continue
 		}
-		if strings.Contains(d.Description, rejectVulnerability) {
+		if strings.Contains(d.Description, rejectVulnerability) || strings.Contains(d.Description, rejectedDoNotUse) || strings.HasPrefix(d.Description, rejectedReason) {
```

Comment:
I don't think we need to limit it to NVD as we may take the status from other sources in the future.

```suggestion
		if d.VulnStatus == rejectedStatus {
			continue
		}
		if strings.Contains(d.Description, rejectVulnerability) || strings.Contains(d.Description, rejectedDoNotUse) || strings.HasPrefix(d.Description, rejectedReason) {
```